packages/autorest.typescript/src/generators/indexGenerator.ts

```diff
@@ -24,6 +24,10 @@ function generateHLCIndex(clientDetails: ClientDetails, file: SourceFile) {
   const { disablePagingAsyncIterators } = getAutorestOptions();
   if (clientDetails.options.hasPaging && !disablePagingAsyncIterators) {
     file.addStatements([`/// <reference lib="esnext.asynciterable" />`]);
```

Comment:
nitpick: unrelated but wanted to highlight it: I don't believe this is needed anymore since our minimum supported node version is 14 and async iterables is part of node 10.

Reply:
Certainly not needed from a platform compat perspective, I think it's more a question of if our tsconfig build target pulls these in or not, have to check. It's a good idea to update the target now that we have a new baseline, so I'll file an issue for it.

Reply:
https://github.com/Azure/azure-sdk-for-js/issues/23641